concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/30 Save, Cancel, and Custom Buttons.md

```diff
@@ -0,0 +1,23 @@
+Form and popup editing modes include built-in Save and Cancel buttons. If you want to execute custom code on click one of these buttons, specify the [onSaving](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onSaving) or [onEditCanceling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditCanceling) event handler correspondingly. Assign `true` to the **e.cancel** parameter to cancel the Save or Cancel operations and keep the form or popup opened.
+
+Specify the **editing.texts**.[cancelRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#cancelRowChanges) and **editing.texts**.[saveRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#saveRowChanges) properties to change texts of the Save and Cancel buttons.
```

Comment:
> Specify **the** `editing.texts.cancelRowChanges` and `editing.texts.saveRowChanges`

This is a case from the new guide on articles. When you enumerate named entities with AND, you need no article.

> to change texts of ...

I will always advise using the word "text" with caution. It's always fine when used as an adjective. When it's used as a noun -- it becomes too vague. When it's used in plural form, "texts" sounds like "articles", "treatises", "books"... I would have used a more precise alternative here. Like "captions", for instance.

> of **the** Save and Cancel buttons

The same incorrect article case.

```suggestion
Specify **editing.texts**.[cancelRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#cancelRowChanges) and **editing.texts**.[saveRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#saveRowChanges) properties to change Save and Cancel button captions.
```